src/visualization/annotation/annotation_base.hpp

```diff
@@ -90,6 +90,9 @@ class EXPORT AnnotationBase : public ml_model_base {
 
   virtual void checkDataSet() = 0;
 
+  /* must be called after construction */
```

Comment:
If it's intended to be called once, and only from the constructor, I wouldn't put it in the public API. It should ideally be private in that case and non-virtual (or protected and virtual, if subclass constructors need to call it directly instead of through their superclass constructor).

Reply:
That's a good point! I should provide it as a private virtual.

Reply:
Ah, I just realized it's a mistake to call virtual function during construction. I will make it non-virtual protected.

Reply:
it's tricky to have virtual functions involved in construction. I will make it non-virtual for this case.

Reply:
@znation as is recorded in #2234. Will be a feature request for @abhishekpratapa. For now, it's okay since base class is pure virtual and derived class directly inherits base. If the inheritance chain grows, then we should consider changing the code structure first. For now, it's irrelevant to this pr.